src/runtime_src/core/pcie/emulation/cpu_em/generic_pcie_hal2/shim.cxx

```diff
@@ -476,9 +476,20 @@ namespace xclcpuemhal2 {
             childArgv[5] = portStream.str().c_str() ;
           }
         }
-        int r = execl(modelDirectory.c_str(), childArgv[0], childArgv[1],
+
+
+        int r = 0;
+
+        if (xclemulation::is_sw_emulation() && xrt_core::config::get_flag_sw_emu_kernel_debug()){// Launch sw_emu device Process in GDB -> Emulation.kernel-dbg =true
```

Comment:
why is is_sw_emulation check inside cpu_em driver?

Reply:
@chvamshi-xilinx , its additional check only, nothing harm.

Reply:
yaa. It might be confusing in future. better to delete :-)

Reply:
sure @chvamshi-xilinx ,will remove this additional check in next PR.

Reply:
Thanks @vboggara-xilinx